airflow/www_rbac/views.py

```diff
@@ -207,6 +206,20 @@ def get_int_arg(value, default=0):
             unfiltered_webserver_dags = [dag for dag in dagbag.dags.values() if
                                          not dag.parent_dag]
 
+        if not filter_dag_ids:
+            orm_dags = {}
+            unfiltered_webserver_dags = []
+        elif filter_dag_ids == {'all_dags'}:
+            orm_dags = {dag.dag_id: dag for dag
+                        in sql_query
+                        .all()}
+        else:
```

Comment:
Seems there's duplicated logic here, the condition should just apply the filter to the query.
```
else:
    sql_query = sql_query.in_(filter_dag_ids)
```
and the `orm_dags` creation can be outside the condition block